app/ide-desktop/lib/dashboard/src/providers/AuthProvider.tsx

```diff
@@ -238,6 +316,25 @@ export default function AuthProvider(props: AuthProviderProps) {
     },
     [userSession?.type]
   )
+  const gtagEventRef = React.useRef(gtagEvent)
```

Comment:
I've seen this snippet 2 times, can we extract this logic into a hook?

Reply:
oh i guess you mean the entire code block? yeah makes sense - next time you might want to shift-click the last line to comment on the entire block though